rflx/pyrflx/typevalue.py

```diff
@@ -209,12 +222,15 @@ def parse(self, value: Union[Bitstring, bytes]) -> None:
         if isinstance(value, bytes):
             value = Bitstring.from_bytes(value)
         value_as_int: int = int(value)
-        if not Number(value_as_int) in self.literals.values():
+        if not Number(value_as_int) in self.literals.values() and not self._type.always_valid:
             raise KeyError(f"Number {value_as_int} is not a valid enum value")
-
-        for k, v in self.literals.items():
-            if v == Number(value_as_int):
-                self._value = str(k)
+        if not Number(value_as_int) in self.literals.values() and self._type.always_valid:
+            self._type.literals["UNKNOWN"] = Number(value_as_int)
+            self._value = "UNKNOWN"
```

Comment:
In bfda76d I  changed _value to a Tuple[str, Number], now the numeric value itself and the associated string representation are stored in _value. This removes the need to change the model.

Reply:
I have the problem that when using the assign method of EnumValue and the Always Valid aspect is true, the string representation and the numeric value have to be passed as opposed to before where only a string was passed. Therefore I changed the signature of assign() to `assign(self, value: Tuple[str, Optional[int]], check: bool = True)`.
If a field is set manually using the set() method of MessageValue there has to be an option to also pass an int value associated with the string in case always valid is true. I thought of the following options:
1. change the signature of set() to 
`set( self, field_name: str, value: Union[bytes, int, Tuple[str, Optional[int]], Sequence[TypeValue], Bitstring] ) -> None`
2. change the signature of set() to 
`set( self, field_name: str, value: Union[bytes, int, str, Sequence[TypeValue], Bitstring], numeric_value_for_enum_fields: Optional[int] = None ) -> None`

Both options are not really great.
1 would have the downside that when setting a field, `None` has to be explicitly written every time, even if "Always Valid" is false
2 is not really self explanatory
What do you think?

Reply:
> when using the assign method of EnumValue and the Always Valid aspect is true, the string representation and the numeric value have to be passed as opposed to before where only a string was passed.

No, the `always_valid` aspect is not relevant for setting a value, as I clarified in https://github.com/Componolit/RecordFlux/pull/219#discussion_r423110842. It __must not__ be possible to assign an unknown value to an EnumValue, even if `always_valid` is true. So there should be no need to change the signature of the `set` and `assign` methods.

Reply:
Only change the `parse` method then? In this case bfda76d and 29929ce should fix this. 
The underlying model is not changed now.

Reply:
Yes, these commits should be a valid solution to the original issue.